### PR TITLE
425 metadata tab edit page

### PR DESCRIPTION
**Pull request by syed-ali-tw**
[Trello](https://trello.com/c/N4sWF2NI/425-metadata-tab-edit-page)

Move Metadata tab to new design system

Screenshot_metadata_tab

### CODE REVIEW

File: test/integration/edition_edit_test.rb [hunk 1]
```diff
@@ -35,4 +31,59 @@ class EditionEditTest < IntegrationTest
...
     assert page.has_text?("Unpublish")
   end
+
+  context "#metadata" do
+    context "#draft" do
+      setup do
+        click_link("Metadata")
+      end
+
+      should "'Metadata' header and an update button" do
```
Comment (mtaylorgds) [inline test/integration/edition_edit_test.rb:41]: This does not read as a sentence. Perhaps use something like "see the 'Metadata' header and an update button" instead.

File: test/integration/edition_edit_test.rb [hunk 2]
```diff
@@ -35,4 +31,59 @@ class EditionEditTest < IntegrationTest
...
+        choose("Welsh")
+        click_button("Update")
+
+        assert find(".gem-c-radio input[value='cy']").checked?
+        assert page.has_text?("Metadata has successfully updated")
+        assert page.has_field?("artefact[slug]", with: "changed-slug")
+      end
+    end
+
+    context "#published" do
```
Comment (mtaylorgds) [inline test/integration/edition_edit_test.rb:65]: Similar to `draft`, `published` is not the name of a method being tested, so we shouldn't have the # prefix in the context name. Might be worth elaborating a bit in the context name as well, for example, "viewing a published edition".

File: test/integration/edition_edit_test.rb [hunk 3]
```diff
@@ -35,4 +31,59 @@ class EditionEditTest < IntegrationTest
...
+          ).id,
+          state: "published",
+          slug: "can-i-get-a-driving-licence",
+        )
+
+        visit edition_path(edition)
+        click_link("Metadata")
+      end
+
+      should "show current value for slug and language as texts" do
```
Comment (mtaylorgds) [inline test/integration/edition_edit_test.rb:81]: Not sure "as texts" is good English. We can probably just omit that bit anyway as it should be clear from reading the assertions that it is testing the displayed UI text.

**Review by mtaylorgds — COMMENTED** [review]
This could be done in a subsequent PR (as part of the same story), but the validation needs work—there should be inline validation messages, and the message at the top should be more readable, and should be in an error summary, with links to the appropriate field.

I thought we were transitioning by content type—so you would only see the design system tabs for help pages and answers, but would still see the old bootstrap UI for other content types—has this changed?

Is there any test coverage for the validation messages?

File: app/models/artefact.rb [hunk 4]
```diff
@@ -270,6 +270,17 @@ def exact_route?
     prefixes.empty?
   end
 
+  def locale_to_language(locale)
```
[the remark this answers is not in the document]
Comment (syed-ali-tw) [reply app/models/artefact.rb:273]: do you mean a helper module?
Although its used for UI at the moment, there might be other uses at well, to me its a translation for a field in the Artefact model.

File: app/helpers/tabbed_nav_helper.rb [hunk 5]
```diff
@@ -29,4 +28,15 @@ def edit_nav_item(label, href, current)
       },
     ]
   end
+
+  def current_tab
+    tab = request.path.split("/") & items
```
[the remark this answers is not in the document]
Comment (syed-ali-tw) [reply app/helpers/tabbed_nav_helper.rb:33]: will rename items as suggested, makes more sense.
I know what you mean though, using & is not so uncommon. Although the name of the tab should always be the last element, I could not find in the documentation that string split would guarantee the order. thus using & felt bit more safer albiet more clever

File: app/helpers/tabbed_nav_helper.rb [hunk 6]
```diff
@@ -29,4 +28,15 @@ def edit_nav_item(label, href, current)
       },
     ]
   end
+
+  def current_tab
```
[the remark this answers is not in the document]
Comment (mtaylorgds) [reply app/helpers/tabbed_nav_helper.rb:32]: This comment was about the method name, which hasn't changed.

File: app/helpers/tabbed_nav_helper.rb [hunk 7]
```diff
@@ -29,4 +28,15 @@ def edit_nav_item(label, href, current)
...
+    current_tab_name = request.path.split("/") & items
```
Comment (mtaylorgds) [inline app/helpers/tabbed_nav_helper.rb:33]: Maybe this one can be `tab_name` (though really it's an array, so should be something like `potential_tab_names`?

Comment (syed-ali-tw) [reply app/helpers/tabbed_nav_helper.rb]: have just kept it as current tab and returning the first element of the array itself.

File: test/integration/edition_edit_test.rb [hunk 8]
```diff
@@ -8,31 +8,90 @@ class EditionEditTest < IntegrationTest
...
+      assert page.has_text?("Edit")
+      assert page.has_text?("Tagging")
+      assert page.has_text?("Metadata")
+      assert page.has_text?("History and notes")
+      assert page.has_text?("Admin")
+      assert page.has_text?("Related external links")
+      assert page.has_text?("Unpublish")
+    end
+
+    context "#metadata" do
```
Comment (mtaylorgds) [inline test/integration/edition_edit_test.rb:39]: I missed this one last time round, but `metadata` is not the name of the method in this context, so would be better as something like "viewing the metadata tab".

File: test/integration/edition_edit_test.rb [hunk 9]
```diff
@@ -8,31 +8,90 @@ class EditionEditTest < IntegrationTest
...
+        click_button("Update")
+
+        assert find(".gem-c-radio input[value='cy']").checked?
+        assert page.has_text?("Metadata has successfully updated")
+        assert page.has_field?("artefact[slug]", with: "changed-slug")
+      end
+    end
+
+    context "when edition is draft" do
+      context "#metadata" do
```
Comment (mtaylorgds) [inline test/integration/edition_edit_test.rb:69]: Similarly `metadata` isn't a method name in this context (see earlier comment).

File: test/integration/edition_edit_test.rb [hunk 10]
```diff
@@ -8,31 +8,90 @@ class EditionEditTest < IntegrationTest
...
+        choose("Welsh")
+        click_button("Update")
+
+        assert find(".gem-c-radio input[value='cy']").checked?
+        assert page.has_text?("Metadata has successfully updated")
+        assert page.has_field?("artefact[slug]", with: "changed-slug")
+      end
+    end
+
+    context "when edition is draft" do
```
Comment (mtaylorgds) [inline test/integration/edition_edit_test.rb:68]: I think this is supposed to be "when edition is published"?

And it should not be within the preceding "when edition is draft" context, but at the same level.

**Review by mtaylorgds — COMMENTED** [review]
A few comments related to the test file structure.

**Review by mtaylorgds — APPROVED** [review]

**Pull request merged**